app/src/main/java/com/example/simbirsoftmobile/presentation/screens/search/events/EventsFragment.kt

```diff
@@ -0,0 +1,143 @@
+        get() = _binding!!
+
+    private val adapter: SearchResultAdapter by lazy { SearchResultAdapter() }
+    private var uiState: UiState<List<SearchResultUI>> = UiState.Idle
+
+    private val compositeDisposable = CompositeDisposable()
+    private val publishSubject: PublishSubject<String> = PublishSubject.create()
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        super.onSaveInstanceState(outState)
+        val currentState = uiState
+        if (currentState is UiState.Success) {
+            outState.putParcelableArrayList(LIST_KEY, ArrayList(currentState.data))
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentEventsBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(
+        view: View,
+        savedInstanceState: Bundle?,
+    ) {
+        super.onViewCreated(view, savedInstanceState)
+        initAdapter()
+
+        if (savedInstanceState != null) {
+            val currentList = getEventListFromBundle(savedInstanceState)
+
+            uiState = UiState.Success(currentList)
+            updateUiState()
+        }
+
+        val disposable = publishSubject.switchMapSingle { query ->
+            EventRepository.searchEvent(query, requireContext())
+        }
+            .subscribeOn(Schedulers.io())
+            .map { it.map { it.toSearchResultUi() } }
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe {
+                uiState = UiState.Success(it)
+                updateUiState()
+            }
+        compositeDisposable.add(disposable)
+    }
+
+    private fun getEventListFromBundle(savedInstanceState: Bundle): List<SearchResultUI> =
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            savedInstanceState.getParcelableArrayList(LIST_KEY, SearchResultUI::class.java)
+                ?.toList()
+                ?: listOf()
+        } else {
+            @Suppress("DEPRECATION")
+            savedInstanceState.getParcelableArrayList<SearchResultUI>(LIST_KEY)?.toList()
+                ?: listOf()
+        }
+
+    private fun updateUiState() {
+        when (val currentState = uiState) {
+            is UiState.Error -> {}
+            UiState.Idle -> {}
+            UiState.Loading -> {}
+            is UiState.Success -> {
+                with(binding) {
+                    emptyResultLayout.visibility = View.GONE
+                    queryResultLayout.visibility = View.VISIBLE
+                    searchResultSizeTV.text =
+                        getString(
+                            R.string.search_result_events_size,
+                            currentState.data.size.toString(),
+                        )
+                }
+                adapter.submitList(currentState.data)
+            }
```

Comment:
Чтобы не прописывать пустые скобки для каждого типа 
```suggestion
        when {
    uiState is UiState.Success -> {
        with(binding) {
                    emptyResultLayout.visibility = View.GONE
                    queryResultLayout.visibility = View.VISIBLE
                    searchResultSizeTV.text =
                        getString(
                            R.string.search_result_events_size,
                            currentState.data.size.toString(),
                        )
                }
                adapter.submitList(currentState.data)
    }
}
```